src/python/buildBinaryLineDocs.py

```diff
@@ -49,17 +50,21 @@ def flush(pending, pendingDocCount, fOut):
     totalLength = len(titleBytes)+len(bodyBytes)+len(randomLabelBytes)+20
     #print('len=%s' % totalLength)
     #print('HERE: %s, offset=%s' % (struct.pack('i', totalLength), fOut.tell()))
-    pending.write(struct.pack('iiiil', len(titleBytes), len(bodyBytes), len(randomLabelBytes), timeSec, msecSinceEpoch))
+    pending.write(struct.pack('iiiiil', len(titleBytes), len(bodyBytes),
+                              len(randomLabelBytes), timeSec, nextId, msecSinceEpoch))
```

Comment:
Hmm why do we need to encode the `id` in the binary file itself?  Can't each thread while decoding block of docs know its base and then increment ID to reconstruct perfectly?

Reply:
Good point, basically we can preallocate a contiguous block of ids within reading thread to each doc so that the id assignment will be deterministic. 

I just think this approach probably will be easier to reason about and understand, what do you think?

Reply:
> I just think this approach probably will be easier to reason about and understand, what do you think?

Well, in general, I try very hard to ensure that this benchmark absolutely minimizes the "outside Lucene indexing" work, i.e. reading lines from text or binary line docs and constructing the documents to index.  So adding something to the file, that takes (even a bit of) storage, requires reading/decoding, that could instead be computed efficiently via thread-private `++` when building the document, makes me nervous.  So I would prefer to do the `++` approach, but I understand this code has become complex with the three different cases it's handling ;)

Reply:
OK, that makes sense to me, I'll try to revise it so that it assigns id without modifying the file